hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaBrokerMessageProducer.java

```diff
@@ -54,21 +55,25 @@ public void send(Message message, CachedTopic cachedTopic, final PublishingCallb
     private ProducerRecord<byte[], byte[]> createProducerRecord(Message message, CachedTopic cachedTopic) {
         String kafkaTopicName = cachedTopic.getKafkaTopics().getPrimary().name().asString();
         Optional<SchemaVersion> schemaVersion = message.<Schema>getCompiledSchema().map(CompiledSchema::getVersion);
+        Optional<SchemaId> schemaId = message.<Schema>getCompiledSchema().map(CompiledSchema::getId);
 
-        Iterable<Header> headers = createRecordHeaders(message.getId(), message.getTimestamp(), schemaVersion);
+        Iterable<Header> headers = createRecordHeaders(message.getId(), message.getTimestamp(), schemaId, schemaVersion);
 
         return new ProducerRecord<byte[], byte[]>(kafkaTopicName, null, null, message.getData(), headers);
     }
 
-    private Iterable<Header> createRecordHeaders(String id, long timestamp, Optional<SchemaVersion> schemaVersion) {
-        return schemaVersion
-                .map(sv -> asList(
-                        kafkaHeaderFactory.messageId(id),
-                        kafkaHeaderFactory.timestamp(timestamp),
-                        kafkaHeaderFactory.schemaVersion(sv.value())))
-                .orElse(asList(
-                        kafkaHeaderFactory.messageId(id),
-                        kafkaHeaderFactory.timestamp(timestamp)));
+    private Iterable<Header> createRecordHeaders(String id, long timestamp, Optional<SchemaId> schemaId, Optional<SchemaVersion> schemaVersion) {
+        Stream<Optional<Header>> headers = Stream.of(
+            Optional.of(kafkaHeaderFactory.messageId(id)),
+            Optional.of(kafkaHeaderFactory.timestamp(timestamp)),
+            schemaVersion.map(sv -> kafkaHeaderFactory.schemaVersion(sv.value())),
+            schemaId.map(sid -> kafkaHeaderFactory.schemaId(sid.value()))
```

Comment:
I've added `schemaIdHeaderEnabled` in Topic configuration.

Reply:
Great, however instead of enable/disable this feature per topic it would be better to do it globally for all topics via Configs.java. Can you provide this change? You are very close to another approve :)

Reply:
Done.